src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+        path_delimiter.LuaString = "/"
+    elseif ui:IsOnWindows() then
+        path_delimiter.LuaString = "\\"
+        --path_delimiter.LuaString = "/" -- apparently Windows can use either! Go figure!
+    end
+    path.LuaString = path.LuaString..path_delimiter.LuaString..filename
+    return path
+end
+
+function harp_config_load()
+    local path = path_set(config_filename)
+    local config_settings = {}
+    local init_settings = {root = 2, accidental = 1, scale = 0, scale_check = 1, chord = 0, chord_check = 0, diagram_check = 1, names_check = 0, partial_check = 0, stack = 1, pedal_lanes = 1, last_notes = "D, C, B, E, F, G, A"}
+    -- This next might not be needed... But doesn't hurt so leaving it in for now...
+    local init_count = 0
+    for i,k in pairs(init_settings) do
+        init_count = init_count + 1
+    end
+    --
+    local file_r = io.open(path.LuaString, "r")
+
+    if file_r == nil then
+        config_settings = init_settings
+        harp_config_save(config_settings)
+        file_r = io.open(path.LuaString, "r")
+    end
+
+    config_settings = get_parameters(path.LuaString, config_settings)
+
+    for key, val in pairs(init_settings) do
+        if config_settings[key] == nil then
+            config_settings[key] = val
+        end
+    end
+    return config_settings
+end -- harp_config_load()
+
+function harp_config_save(config_settings)
+    local path = path_set(config_filename)
+    local file_w = io.open(path.LuaString, "w")
+    for key, val in pairs(config_settings) do
+        file_w:write(key.." = "..val.."\n")
+    end
+    file_w:close()
+end -- harp_config_save()
+
+function getUsedFontName(standard_name)
+    local font_name = standard_name
+    if string.find(os.tmpname(), "/") then
+        font_name = standard_name
+    elseif string.find(os.tmpname(), "\\") then
+        font_name = string.gsub(standard_name, "%s", "")
+    end
+    return font_name
+end
+
+function get_def_mus_font()
+    local fontinfo = finale.FCFontInfo()
+    if fontinfo:LoadFontPrefs(finale.FONTPREF_MUSIC) then
+        return getUsedFontName(fontinfo:GetName())
+    end
+end
+
+---
+function harp()
+    finenv.RetainLuaState = true
+    local partial = false
+    local changes = false
+    local stack = true
+    local pedal_lanes = true
+    local direct = false
+    local override = false
+    local context =
+    {
+        window_pos_x = window_pos_x or nil,
+        window_pos_y = window_pos_y or nil
+    }
+    local SMuFL = library.is_font_smufl_font(nil)
+    harpstrings = {}
+    diagram_string = finale.FCString()
+    description = finale.FCString()
+    local changes_str = finale.FCString()
+    changes_str.LuaString = ""
+    local default_music_font = get_def_mus_font()
+    local diagram_font = "^fontTxt("..default_music_font..")"
+    --
+    local flat_char = ""
+    local nat_char = ""
+    local sharp_char = ""
+    local cross_char = ""
+    local desc_prefix = finale.FCString()
```

Comment:
When RGP Lua 0.63 comes out (soon I think), this needs to be changed to, e.g.:

```lua
local flat_char = utf8.char(0xe360) -- standard SMuFL flat
```
Embedding SMuFL characters directly in the text file is risky because text fonts used by most editor will not be able to display them, so you can't tell from the code what the character is.

Reply:
Added the utf 8 codes but commented out, waiting for 0.63